cloud/vmware/vsphere_guest.py

```diff
@@ -689,6 +692,10 @@ def deploy_template(vsphere_client, guest, resource_pool, template_src, esxi, mo
                 cloneArgs["linked"] = True
                 cloneArgs["snapshot"] = snapshot_to_clone
 
+            if vm_extra_config.get("folder") is not None:
```

Comment:
I _think_ there may be an edge case here where `vm_extra_config.get("folder")` returns an empty string (for example, if the user supplied an empty string). Will this feature degrade gracefully in this scenario?

If not, another option might be ` if not vm_extra_config.get("folder"):` which would evaluate as a truthy/falsey, vs specifically testing for `None`.